stix2/properties.py

```diff
@@ -171,9 +182,7 @@ def __init__(self, type):
     def clean(self, value):
         if not value.startswith(self.required_prefix):
             raise ValueError("must start with '{0}'.".format(self.required_prefix))
-        try:
-            uuid.UUID(value.split('--', 1)[1])
-        except Exception:
+        if not ID_REGEX.match(value):
             raise ValueError("must have a valid UUID after the prefix.")
```

Comment:
Same here. We could update the message to UUIDv4 to prevent confusion on why other UUID versions raise an error.

Reply:
Agreed! And honestly, the message might need to be more general, since the regex may also catch things unrelated to the UUID.